src/Quahog/Client.php

```diff
@@ -257,14 +261,34 @@ private function _sendCommand($command)
     /**
      * A wrapper to cleanly read a response from clamd.
      *
+     * @param bool $removeId
+     * @param string $readUntil
      * @return string
+     * @throws ConnectionException
      */
-    private function _receiveResponse()
+    private function _receiveResponse($removeId = false, $readUntil = "\n")
     {
-        $result = $this->_socket->read(4096);
+        $result = null;
+        $readUntilLen = strlen($readUntil);
+        do {
+            if ($this->_socket->selectRead($this->_timeout)) {
+                $rt = $this->_socket->read(4096, PHP_NORMAL_READ);
```

Comment:
What was the reason for the switch to `PHP_NORMAL_READ`? By default, the socket uses `PHP_BINARY_READ` and even though no test failed, this may be a BC break. The implemented tests mock the socket away, so maybe it is a good idea to start with some integration tests and compare normal with binary. Integration tests should be in a new test class.

@jonjomckay You already added clamAV installation in travis-config, so I guess you were planning to introduce integration tests in the future anyways?